launcher/ui/pages/instance/StoragePage.cpp

```diff
@@ -0,0 +1,159 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "StoragePage.h"
+#include <QMessageBox>
+#include <QStorageInfo>
+#include <QTabBar>
+#include "ui_StoragePage.h"
+
+qint64 getDirectorySize(QString path)
+{
+    QDirIterator it(path, QDirIterator::Subdirectories);
+    qint64 fileSizeTotal = 0;
+    while (it.hasNext()) {
+        it.next();
+        fileSizeTotal += it.fileInfo().size();
+    }
+    return fileSizeTotal;
+}
+
+void clearDirectoryInner(QString path)
+{
+    QDir(path).removeRecursively();
+}
+
+StoragePage::StoragePage(BaseInstance* inst, QWidget* parent) : QWidget(parent), ui(new Ui::StoragePage), m_inst(inst)
+{
+    ui->setupUi(this);
+
+    m_confirmation_box = new QMessageBox(this);
+    m_confirmation_box->setWindowTitle("Confirmation");
+    m_confirmation_box->setIcon(QMessageBox::Warning);
+    m_confirmation_box->setText("Are you sure you want to proceed?");
+    m_confirmation_box->setStandardButtons(QMessageBox::Yes);
+    m_confirmation_box->addButton(QMessageBox::No);
+    m_confirmation_box->setDefaultButton(QMessageBox::No);
+
+    m_series = new QtCharts::QPieSeries(this);
+    m_series->setLabelsVisible();
+    m_series->setLabelsPosition(QtCharts::QPieSlice::LabelInsideHorizontal);
+    m_chart_view = new QtCharts::QChartView(this);
+    m_chart = new QtCharts::QChart();
+    m_chart->addSeries(m_series);
+    m_chart->setBackgroundVisible(false);
+    m_chart->setMargins(QMargins(0, 0, 0, 0));
+    m_chart->legend()->setAlignment(Qt::AlignLeft);
+    m_chart_view->setRenderHint(QPainter::Antialiasing);
+
+    ui->verticalLayout->addWidget(m_chart_view);
+    ui->retranslateUi(this);
+
+    updateCalculations();
+
+    connect(ui->button_goto_resouce_packs, &QPushButton::clicked, this, [&] { m_container->selectPage("resourcepacks"); });
+    connect(ui->button_goto_mods, &QPushButton::clicked, this, [&] { m_container->selectPage("mods"); });
+    connect(ui->button_goto_worlds, &QPushButton::clicked, this, [&] { m_container->selectPage("worlds"); });
+    connect(ui->button_goto_screenshots, &QPushButton::clicked, this, [&] { m_container->selectPage("screenshots"); });
+    connect(ui->button_goto_other_logs, &QPushButton::clicked, this, [&] { m_container->selectPage("logs"); });
+
+    connect(ui->button_clear_screenshots, &QPushButton::clicked, this, &StoragePage::handleClearScreenshotsButton);
+    connect(ui->button_clear_logs, &QPushButton::clicked, this, &StoragePage::handleClearLogsButton);
+    connect(ui->button_clear_all, &QPushButton::clicked, this, &StoragePage::handleClearAllButton);
+}
+
+StoragePage::~StoragePage()
+{
+    delete m_chart;
```

Comment:
Use setParent after allocating the object instead of manually destroying it here